src/data/set/pairwise.lean

```diff
@@ -192,6 +192,17 @@ lemma pairwise.pairwise_on (h : pairwise r) (s : set α) : s.pairwise_on r := λ
 lemma pairwise_disjoint_fiber (f : ι → α) : pairwise (disjoint on (λ a : α, f ⁻¹' {a})) :=
 set.pairwise_on_univ.1 $ set.pairwise_on_disjoint_fiber f univ
 
+lemma pairwise_subtype_iff_pairwise_on {α : Type*} (s : set α) (r : α → α → Prop) :
+  pairwise (λ (x : s) (y : s), r x y) ↔ s.pairwise_on r :=
```

Comment:
This conflicts with #10035

Reply:
But also, this lemma is great, and I think we should provide dot notation too for the reverse direction, as `set.pairwise.subtype_pairwise`

Reply:
I'll merge master once #10035 has landed.